tst/worksheet.tst

```diff
@@ -0,0 +1,62 @@
+#############################################################################
+##
+##  AutoDoc package
+##
+##  Test the behavior of AutoDocWorksheet
+##
+##  Copyright 2018
+##    Contributed by Glen Whitney, studioinfinity.org
+##
+## Licensed under the GPL 2 or later.
+##
+#############################################################################
+
+gap> START_TEST( "AutoDoc package: worksheet.tst" );
+gap> tmpdir := DirectoryTemporary();;
+gap> SetInfoLevel( InfoGAPDoc, 0 );
+gap> AutoDocWorksheet( "tst/worksheet.tst", rec( dir := tmpdir ) );
+gap> chap1 := Filename( Directory(tmpdir), "chap1.html" );;
+gap> IsReadableFile(chap1);
+true
+gap> ReadAll(InputTextFile(chap1));
```

Comment:
I am not quite comfortable with this solution for testing the output. For once, it is prone to break when GAPDoc gets an update that changes the generated HTML. "Luckily", past experience tell us that GAPDoc updates are not exactly frequent.

But more importantly, this solution makes it very difficult to debug any test failures, because diff'ing the below text against the actually generated file is difficult. 

IMHO a better approach would be to store the expected solution as its own file, e.g. `tst/worksheet.expected/chap1.html`; then put the output of the test not into a random temp dir, but rather into `tst/worksheet.actual/` (which could be added to the `.gitignore`). Then the test here could luck roughly like this:
```
gap> ReadAll(InputTextFile(chap1_actual)) = ReadAll(InputTextFile(chap1_expected));
true
```
Then, if there is a test failure, I can just run `diff -u tst/worksheet.expected/chap1.html tst/worksheet.actual/chap1.html` to see what's going on.

Reply:
Indeed, in working with this approach I ran into exactly the same concerns, and in fact in the latest large pull request implements (among various things) almost exactly the changes you advocate; I switched to testing the XML rather than html, since AutoDoc's job, really, is just to generate the XML; and I added reference files and compared with them, rather than having the expected results in the .tst file.

However, there were three slight differences in my approach in that pull request as compared to what you are describing in this comment.

1) I still wrote the results into a tmpdir. I did not want to assume that the package directory was writable by the user id running the tests. (My mental model was one user on a multi-user machine who wants to run the tests against the installed common GAP installation; presumably in that case the user cannot write to the pkg directories of the common installation.) So my plan in the revised pull request will be to test if a directory of the form tst/worksheets/test1.actual can be created and is writable, and if so run the test there, but if not, fall back to a tmpdir.  If that doesn't seem like a good approach, let me know.

3) Rather than doing ReadAll on both and comparing the strings, I Exec()d a diff command to write its output to a file, and then did a ReadAll on the diffs. That way, if the test fails, the diffs are right there to see in the test report.

3) I deliberately put a small, one-line difference between the expected and actual, because to me, seeing that expected difference was more reassuring than completely silent output, which I worried could mask some issue where the entire operation went south and produced nothing. But I agree it is easier to deal with if the expected files don't have to be modified from AutoDoc's output when updating them. So I will switch to using the exact expected files, and use "diff -s -c" so that I have a message that the files are identical to check against for the success case. Again, if you would like any adjustments to that approach, just let me know.